lib/exercises.rb

```diff
@@ -1,29 +1,48 @@
-
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 
-def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+def grouped_anagrams(strings) 
+  h = {}
+  return [] if strings.nil? || strings.empty? || !strings.is_a?(Array)
+  strings.each do |str|   
+    key = str.chars.sort.join
+    if h.include?(key)
```

Comment:
This can be simplified by

```suggestion
    unless h[key].nil?
```